1-Vanilla/main.js

```diff
@@ -1,231 +1,245 @@
       id: 11,
     },
     {
       text: "Infra webinar at 3",
       status: "unfinished",
-      isEditMode: false,
       id: 12,
     },
     {
       text: "Pay the bills",
       status: "unfinished",
-      isEditMode: false,
       id: 13,
     },
     {
       text: "Contribute code to open source",
       status: "unfinished",
-      isEditMode: false,
       id: 14,
     },
     {
       text: "Finish Velo site",
       status: "unfinished",
-      isEditMode: false,
       id: 15,
     },
   ];
-  
+
+  const editModeItems = [];
+
+  const ul = document.getElementById("todoList");
+
   const loadToDoList = (list) => {
-    const ul = document.getElementById("todoList");
     ul.innerHTML = "";
     list.forEach((item) => {
       ul.appendChild(createItem(list, item));
     });
   };
-  
+
   const addTodo = () => {
-    const inputText = document.getElementById("addTodo").value;
-    list.push({
-      text: inputText,
+    const inputText = document.getElementById("addTodo");
+    const newTodo = {
+      text: inputText.value,
       status: "unfinished",
       id: list.length + 1,
-    });
-    loadToDoList(list);
+    };
+    list.push(newTodo);
+    ul.insertBefore(createItem(list, newTodo), ul.firstChild);
+    inputText.value = ""; //clear input text
   };
-  
+
+  // Remove the todo item from the list and from the DOM
   const removeTodo = (list, item) => {
     list = list.filter(function (ele) {
       return ele.id != item.id;
     });
-  
-    loadToDoList(list);
+    const li = document.getElementById(item.id);
+    li.remove();
+  };
+
+  function removeElement(array, elem) {
+    var index = array.indexOf(elem);
+    if (index > -1) {
+      array.splice(index, 1);
+    }
+  }
+
+  const renderComponent = (elem, curr, modifiedElem) => {
+    elem.firstChild.replaceChild(curr, modifiedElem);
   };
-  
-  const editTodo = (list, item) => {
-    list = list.map((ele) => {
-      if (ele.id === item.id) {
-        ele.isEditMode = !ele.isEditMode;
+
+  const editTodo = (item) => {
+    const li = document.getElementById(item.id);
+    const isFound = editModeItems.some((element) => {
+      if (element.id === item.id) {
+        return true;
       }
-      return ele;
     });
-    loadToDoList(list);
+
+    const elemToChange = li.firstChild.children[1];
+    if (isFound) {
+      removeElement(editModeItems, item);
+      const todo = document.createElement("span");
+      const editedText = document.getElementById(`${item.id}-editTodo`);
+      todo.innerText = editedText.value;
+      todo.classList.add("todo-text");
+      if (item.status === "finished") {
+        todo.classList.add("finished-todo");
+        li.firstChild.checked = true;
+      }
+      renderComponent(li, todo, elemToChange);
+    } else {
+      editModeItems.push(item);
+      const inputText = document.createElement("input");
+      inputText.type = "text";
+      inputText.value = item.text;
+      inputText.classList.add("todo-text");
```

Comment:
`classList` can accept multiple strings as argument, e.g:
`inputText.classList.add("todo-text", "edito-todo")`

Reply:
Ok, fixed it, thanks.